src/LoanAgent.sol

```diff
@@ -49,73 +38,166 @@ contract LoanAgent {
+            })
+        );
+    }
+
+    // miner can repay the loan and change the owner to itself
+    function repayLoanAndChangeOwner() external payable onlyRealOwner {
+        accumulateDebt();
+        uint expectedReturnAmount = loanAgentInfo.expectedReturnAmount;
+        require(
+            msg.value + totalOwnerCredit >= expectedReturnAmount,
+            "Insufficient repay amount"
+        );
+        require(
+            payable(aFil).send(expectedReturnAmount),
+            "Fail to return users stakes to aFIL"
+        );
 
-    // miner can repay the loan and change the owner to itself, LoanAgent will selfdestruct and transfer the funds to LendingVault
-    function repayLoanAndChangeOwner(address ownerAddress) external {}
+        address realOwner = addressInfo.realOwner;
+        MinerAPI.changeOwnerAddress(
+            addressInfo.filActorId,
+            FilAddresses.fromEthAddress(realOwner)
+        );
 
-    function changeOwnerBackToReal() external {}
+        require(
+            payable(realOwner).send(address(this).balance),
+            "Fail to return remaining balance to real owner"
+        );
+        totalOwnerCredit = 0;
+        status = NodeStatus.Terminated;
+    }
 
     function getBalanceInfo()
         external
-        view
+        // view // somehow these get functions in MinerAPI is not `view`: https://github.com/Zondax/filecoin-solidity/issues/359
         returns (
-            uint256 availableBalance,
-            uint256 lockedReward,
-            uint256 initialPledgeCollateral
-        ) {}
+            CommonTypes.BigInt memory availableBalance,
+            uint initialPledgeCollateral,
+            MinerTypes.GetVestingFundsReturn memory lockedReward
+        )
+    {
+        availableBalance = MinerAPI.getAvailableBalance(addressInfo.filActorId);
+        initialPledgeCollateral = loanAgentInfo.nodeOwnerStakes;
+        lockedReward = MinerAPI.getVestingFunds(addressInfo.filActorId);
+    }
 
     function getStakeInfo()
         external
         view
-        returns (uint256 nodeOnwerStakes, uint256 usersStakes) {}
-
-    function getNodeOwner() external view returns (address) {}
+        returns (uint nodeOnwerStakes, uint usersStakes)
+    {
+        nodeOnwerStakes = loanAgentInfo.nodeOwnerStakes;
+        usersStakes = loanAgentInfo.usersStakes;
+    }
 
-    // Can only be callled by oracle contract or stakepool factory
-    function updateNodeStatus(NodeStatus status) external {}
+    function getNodeOwner()
+        external
+        // view // somehow these get functions in MinerAPI is not `view`: https://github.com/Zondax/filecoin-solidity/issues/359
+        returns (
+            MinerTypes.GetOwnerReturn memory
+        )
+    {
+        return MinerAPI.getOwner(addressInfo.filActorId);
+    }
 
-    function destructStakePool() external onlyFactory {}
+    // Can only be callled by oracle contract or loanAgent factory
```

Comment:
currentStatusTimestamp is a variable that keeps time at which the status was updated. This variable is used by oracles to determine the deadline of requests in different stages